utilities/pipelines/staticValidation/compliance/helper/helper.psm1

```diff
@@ -341,3 +341,83 @@ function Get-WorkflowEnvVariablesAsObject {
 
     return $envVars
 }
+
+<#
+Get a list of all versioned parents of a module
+
+.DESCRIPTION
+Get a list of all versioned parents of a module. The function will recursively search the parent directories of the given path until it finds a directory containing a version.json file or reaches the root path.
+The function will return a list of all the directories in the parent hierarchy that contain a version.json file, including the given path.
+The function will return the list in the order from the root path to the given path.
+
+.PARAMETER Path
+Mandatory. The path of the module to search for versioned parents.
+
+.PARAMETER RootPath
+Optional. The root path to stop the search at. The function will not search above this path.
+
+.PARAMETER Filter
+Optional. Only include module folders in the list (i.e., modules with a `main.json` file), or versioned modules folders (i.e., modules with `version.json` files). The default is to include all folders).
+
+.EXAMPLE
+Get-ParentFolderPathList -Path 'C:/bicep-registry-modules/avm/res/storage/storage-account/blob-service/container/immutability-policy'
+
+Get all parent folders of the 'immutability-policy' folder up to 'res'. Returns
+
+- <repoPath>\avm\res\storage
+- <repoPath>\avm\res\storage\storage-account
+- <repoPath>\avm\res\storage\storage-account\blob-service
+- <repoPath>\avm\res\storage\storage-account\blob-service\container
+
+.EXAMPLE
+Get-ParentFolderPathList -Path '.\avm\res\storage\storage-account\blob-service\container\immutability-policy' -RootPath 'C:/bicep-registry-modules/avm/res' -Filter 'OnlyVersionedModules'
+
+Get all versioned parent module folders of the 'immutability-policy' folder up to 'res'. Returns, e.g.,
+- <repoPath>\avm\res\storage\storage-account
+#>
+
+.EXAMPLE
+Get-VersionedParentPathList -Path 'C:/bicep-registry-modules/avm/res/storage/storage-account/blob-service/container' -RootPath 'C:/bicep-registry-modules/avm/res'
+
+...
+#>
+function Get-ParentFolderPathList {
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [string] $Path,
+
+        [Parameter(Mandatory = $false)]
+        [string] $RootPath = $repoRootPath,
```

Comment:
Given what you can do with it, I wonder if it would make sense to rename it to something else. Lile 'UpperBoundPath' or something similar

Reply:
done